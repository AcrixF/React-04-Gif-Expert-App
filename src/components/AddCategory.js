import React, { useState } from 'react';

const AddCategory = () => {

    const [inputValue, setInputValue] = useState('Brenda Marlen')

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(e);
    }

    return(
        <form onSubmit={handleSubmit}>
            <input 
                type="text" 
                value = {inputValue}
                onChange = { handleInputChange }/>
        </form>
    );
};

export default  AddCategory;