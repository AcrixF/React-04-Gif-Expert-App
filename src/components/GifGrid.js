import React, { useState, useEffect} from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const [images, setimages] = useState([]);

     useEffect(() => {
        getGifs();
    }, [])

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?api_key=K18TQHOfMMrCINMhsYf6BKbgeq99KP0z&q=Rick and Morty&limit=10';
        const response = await fetch(url);
        const { data } = await response.json();

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        })
        console.log(data);

        setimages(gifs);
    }

    return(
    <>
        <h3>{ category }</h3>
         <div className="card-grid">
            {
                images.map((img) => 
                    <GifGridItem 
                        key={ img.id } 
                        {...img}
                    /> 
                )
            }
        </div>
    </>)
};

