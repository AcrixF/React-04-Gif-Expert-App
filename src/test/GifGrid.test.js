import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../components/GifGrid';
import { useFetchGifs } from '../hooks/useFetchGifs';

jest.mock('../hooks/useFetchGifs')

describe('Testing the <GifGrid/> component', () => {

    
    const category = 'Hunter X Hunter';

    test('should show the <GifGrid/> Shapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true  
        });

        let wrapper = shallow(<GifGrid category= {category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should show the List of Item  when the useFetchGifs is done', () => {
        
        const gifs = [
            {
                id: 'ABC',
                url: 'localhost:8080/goku.gif',
                title: 'Dragon Ball'
            },
            {
                id: 'ABCD',
                url: 'localhost:8080/goku.gif',
                title: 'Dragon Ball Super'
            }
    
        ] 

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false  
        });

        const wrapper = shallow(<GifGrid category= {category}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
    
    

});