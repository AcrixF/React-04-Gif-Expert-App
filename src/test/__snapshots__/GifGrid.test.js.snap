// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the <GifGrid/> component should show the <GifGrid/> Shapshot 1`] = `
<Fragment>
  <h3>
    Hunter X Hunter
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing the <GifGrid/> component should show the List of Item  when the useFetchGifs is done 1`] = `
<Fragment>
  <h3>
    Hunter X Hunter
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Dragon Ball"
      url="localhost:8080/goku.gif"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="Dragon Ball Super"
      url="localhost:8080/goku.gif"
    />
  </div>
</Fragment>
`;
